pkg/selector/selector.go

```diff
@@ -249,9 +195,86 @@ func (itf Selector) rawFilter(filters Filters) ([]*instancetypes.Details, error)
 	for it := range instanceTypes {
 		filteredInstanceTypes = append(filteredInstanceTypes, it)
 	}
+
 	return sortInstanceTypeInfo(filteredInstanceTypes), nil
 }
 
+// sortInstanceTypeInfo will sort based on instance type info alpha-numerically
+func sortInstanceTypeInfo(instanceTypeInfoSlice []*instancetypes.Details) []*instancetypes.Details {
+	if len(instanceTypeInfoSlice) < 2 {
+		return instanceTypeInfoSlice
+	}
+	sort.Slice(instanceTypeInfoSlice, func(i, j int) bool {
+		iInstanceInfo := instanceTypeInfoSlice[i]
+		jInstanceInfo := instanceTypeInfoSlice[j]
+		return strings.Compare(aws.StringValue(iInstanceInfo.InstanceType), aws.StringValue(jInstanceInfo.InstanceType)) <= 0
+	})
+	return instanceTypeInfoSlice
+}
+
+// OutputInstanceTypes accepts a list of instance types details, a number of max results, and an output flag
+// and returns a list of formatted strings representing the passed in intance types with at most maxResults number
+// of results. The format of the strings is determined by the output flag. The number of truncated results
+// is also returned.
+// Accepted output flags: "table", "table-wide", "one-line", "simple", "verbose".
+func (itf Selector) OutputInstanceTypes(instanceTypes []*instancetypes.Details, maxResults int, outputFlag *string) ([]string, int, error) {
```

Comment:
I don't think this makes sense to be nested under `Selector` since it's not using the accessor.  It might make more sense to move the formatting funcs to a new file. 

Also, `outputFlag` is more of a CLI concern, if you want to have a switchable func, I think you should at least type the output option.  It could still be a type alias to a string.  Once this is decoupled completely from the `Selector` it might make sense to make these discreet functions rather than combining.

Reply:
I do see how it might be confusing for the output formatting function to be associated with the selector when the selector itself is not needed for formatting. My initial thought process was that it would ease the users' workflow by reducing the amount of import statements, but I do think it would make more sense to have output functions exist in `outputs.go`.

I do also see how the use of a flag for library users seems unnecessary when they could just call the output functions themselves. For this reason, maybe it would be best for me to use `OutputInstanceFunctions` as a private function used exclusively in `main.go` by the CLI tool, while just expecting library users to call the appropriate output functions if they wish to format instance types as a string. I could also expose the `truncateResults()` function in `selector.go` to the users so that they still have the option to truncate results before output. `truncateResults()` could also exist in `ouputs.go` since it would only be used for output.

Reply:
yeah, all that sounds good. I would advocate for `truncateResults` to stay w/ outputs. If truncate applied to `selector`, it gives users the false idea that the filtering logic is paginated. In reality, it's not possible (at least easily) to paginate a filtering request that is also accurate.